### PR TITLE
Refacto: scrapping sitemap

### PR DESCRIPTION
**Pull request by polomarcus**
- Hands on [advtools lib](https://advertools.readthedocs.io/en/master/advertools.sitemaps.html#)
- Add tests to parse some sitemap.xml locally using a NGINX container to serve on localhost
- Add `url` attribute to PG containing the link of the news

Readme : 
* https://github.com/polomarcus/barometre-medias-quota-climat/tree/refacto/sitemap_parsing#docker

### CODE REVIEW

File: test/sitemap/test_scrap_sitemap.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from quotaclimat.data_ingestion.ingest_db.ingest_sitemap_in_db import get_sitemap_list
+
+def test_get_sitemap_list():
+    sitemap = list(get_sitemap_list())[0]
+    # locally we test only a few items
+    sitemap_url = sitemap
+    sitemap_url == "http://nginxtest:80/sitemap_news_figaro_3.xml"
+
+def test_query_one_sitemap_and_transform():
+    sitemap_config = get_sitemap_list()
```
Comment (polomarcus) [inline test/sitemap/test_scrap_sitemap.py:30]: Main part of this PR, be able to parse a sitemap.xml from a localhost and be sure about its format.

File: quotaclimat/data_processing/sitemap/sitemap_processing.py [hunk 2]
```diff
@@ -59,93 +58,6 @@ def filter_df(df, date_lower_bound, date_upper_bound, keywords):
     return df_between_two_dates_kw
 
 
-def scan_for_duplicates_and_overwrite_the_history(
```
Comment (polomarcus) [inline quotaclimat/data_processing/sitemap/sitemap_processing.py:62]: the json way is not used anymore

File: quotaclimat/data_ingestion/scrap_sitemap.py [hunk 3]
```diff
@@ -91,107 +110,31 @@ def query_one_sitemap_and_transform(media: str, sitemap_conf: Dict) -> pd.DataFr
...
-    return df
-
-
-def sanity_check():
-    """Checks if the data is correct, input a df"""
-    # TODO data model here
-    return
-
-
-def write_df(df: pd.DataFrame, media: str):
```
Comment (polomarcus) [inline quotaclimat/data_ingestion/scrap_sitemap.py:134]: the json way not used anymore

File: mockwebsite/sitemap_news_figaro_3.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (polomarcus) [inline mockwebsite/sitemap_news_figaro_3.xml:1]: served on http://localhost:8000/sitemap_news_figaro_3.xml thanks to the nginx container of the docker-compose.yml

File: docker-compose.yml [hunk 5]
```diff
@@ -41,6 +43,14 @@ services:
       - ./app.py:/app/app.py
     depends_on:
       - postgres_db
+      - nginxtest
+
+  nginxtest: # to test locally webpages
```
Comment (polomarcus) [inline docker-compose.yml:48]: http://localhost:8000/sitemap_news_figaro_3.xml

File: docker-compose.yml [hunk 6]
```diff
@@ -26,8 +27,9 @@ services:
     build:
       context: ./
       dockerfile: Dockerfile
-    entrypoint: ["poetry", "run", "pytest", "--cov", "test", "test/"]
+    entrypoint: ["poetry", "run", "pytest", "-v", "--cov", "test", "test/"]
```
Comment (polomarcus) [inline docker-compose.yml:30]: code coverage + test

File: quotaclimat/data_ingestion/config_sitmap.py [hunk 7]
```diff
@@ -84,11 +89,11 @@
         "regex_section": None,  # articles en year/month/day/titre
         "filter_date_label": "lastmod",
     },
-    "letelegramme": {
-        "sitemap_index": None,
-        "sitemap_url": "https://www.letelegramme.fr/metasitemap_news.xml",
-        "regex_section": r"^https:\/\/www\.letelegramme\.fr\/(?P<section>[\/\w-]*)\/.+$",
-    },
+    # "letelegramme": { #TODO fix me
```
Comment (estellerambier) [inline quotaclimat/data_ingestion/config_sitmap.py:92]: 😎

File: quotaclimat/data_ingestion/scrap_sitemap.py [hunk 8]
```diff
@@ -91,107 +110,31 @@ def query_one_sitemap_and_transform(media: str, sitemap_conf: Dict) -> pd.DataFr
         pd.DataFrame
     """
     try:
+        logging.info("Parsing %s with %s" % (media, sitemap_conf["sitemap_url"]))
+        #@see https://advertools.readthedocs.io/en/master/advertools.sitemaps.html#news-sitemaps
         temp_df = adv.sitemap_to_df(sitemap_conf["sitemap_url"])
     except AttributeError:
         logging.error(
             "Sitemap query error for %s: %s does not match regexp."
```
Comment (estellerambier) [inline quotaclimat/data_ingestion/scrap_sitemap.py]: Maybe the log can be more specific, as this is caught within this method

**Review by estellerambier — COMMENTED** [review]
Looks good to me. 
Could you consider adding the url, as you did on your project? I found it relevant

**Pull request merged**